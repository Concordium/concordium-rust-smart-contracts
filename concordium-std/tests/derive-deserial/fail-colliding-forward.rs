//! Ensure `derive(Deserial)` fails when 'forward' attributes are colliding.
use concordium_std::*;

#[derive(Deserial)]
#[concordium(repr(u8))]
enum Count {
    One {
        field: u32,
    },
    #[concordium(forward = [5, 6])]
    Two(Inner),

    #[concordium(forward = 5)]
    Three(Inner),
}

#[derive(Deserial)]
#[concordium(repr(u8))]
enum Inner {
    #[concordium(tag = 5)]
    Alpha {
        balance: u32,
    },
    #[concordium(tag = 6)]
    Beta(u16),
}

fn main() {}
