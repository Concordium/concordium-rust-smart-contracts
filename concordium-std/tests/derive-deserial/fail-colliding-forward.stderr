error: The tag of 'Three' collides with the tag of 'Two'.
  --> tests/derive-deserial/fail-colliding-forward.rs:13:28
   |
13 |     #[concordium(forward = 5)]
   |                            ^

error: The tag of 'Two' collides with the tag of 'Three'.
  --> tests/derive-deserial/fail-colliding-forward.rs:10:29
   |
10 |     #[concordium(forward = [5, 6])]
   |                             ^
